### PR TITLE
TINY-10101: Fix Safari/Firefox not showing border when focus on iframe

### PR DESCRIPTION
**Pull request by shanmen-tiny**
Related Ticket: TINY-10101

Description of Changes:
* Borders around the div instead of the iframe, there are some issues on Safari and Firefox the border still shows while the focus has shifted
* Created a channel within the dialog and broadcast the new focus element on `dialogFocusShiftedChannel`, `iframe` in dialog has `Receiving.config` to compare if the iframe has focus to remove the focussed class

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):

### CODE REVIEW

**Review by MitchCTiny — APPROVED** [review]

File: modules/tinymce/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 - On Safari and Firefox browsers, scroll positions were not always maintained when updating the content of a `streamContent: true` iframe dialog component. #TINY-10078 #TINY-10109
 - Scrolling behavior was inconsistent when updating a `streamContent: true` iframe dialog component with content lacking an HTML document type declaration. #TINY-10110
+- On Safari and Firefox browsers, the border around the `iframe` dialog component was not visible. #TINY-10101
```
Comment (mkzhx) [inline modules/tinymce/CHANGELOG.md:15]:
```suggestion
- On Safari and Firefox browsers, the border around the `iframe` dialog component was not highlighted when focused. #TINY-10101
```

File: modules/tinymce/src/themes/silver/main/ts/ui/dialog/BodyPanel.ts [hunk 2]
```diff
@@ -55,7 +56,15 @@ const renderBodyPanel = (spec: BodyPanelSpec, dialogData: Dialog.DialogData, bac
           },
           Fun.identity
         )
-      })
+      }),
+      AddEventsBehaviour.config('dialog-body-panel', [
+        // TINY-10101: This is to cater the case where clicks are made into the dialog instead using keyboard navigaton, as FocusShifted would not be triggered in that case.
```
Comment (mkzhx) [inline modules/tinymce/src/themes/silver/main/ts/ui/dialog/BodyPanel.ts:61]:
```suggestion
        // TINY-10101: This is to cater for the case where clicks are made into the dialog instead using keyboard navigation, as FocusShifted would not be triggered in that case.
```

File: modules/tinymce/src/themes/silver/main/ts/ui/dialog/Dialogs.ts [hunk 3]
```diff
@@ -10,6 +10,7 @@ import Env from 'tinymce/core/api/Env';
 import { UiFactoryBackstageProviders } from '../../backstage/Backstage';
 import * as HtmlSanitizer from '../core/HtmlSanitizer';
 import * as NavigableObject from '../general/NavigableObject';
+import { dialogFocusShiftedChannel } from '../window/DialogChannels';
```
Comment (mkzhx) [inline modules/tinymce/src/themes/silver/main/ts/ui/dialog/Dialogs.ts:13]: Scoped imports might be preferable here
```suggestion
import * as DialogChannels from '../window/DialogChannels';
```

File: modules/tinymce/src/themes/silver/main/ts/ui/dialog/Dialogs.ts [hunk 4]
```diff
@@ -175,6 +176,11 @@ const renderDialog = (spec: DialogSpec): SketchSpec => {
           // Using just `run` instead will cause an infinite loop as `focusIn` would fire a `focusin` which would then get responded to and so forth.
           AlloyEvents.runOnSource(NativeEvents.focusin(), (comp, _se) => {
             Blocking.isBlocked(comp) ? Fun.noop() : Keying.focusIn(comp);
+          }),
+          AlloyEvents.run<SystemEvents.AlloyFocusShiftedEvent>(SystemEvents.focusShifted(), (comp, se) => {
+            comp.getSystem().broadcastOn([ dialogFocusShiftedChannel ], {
```
Comment (mkzhx) [inline modules/tinymce/src/themes/silver/main/ts/ui/dialog/Dialogs.ts:181]:
```suggestion
            comp.getSystem().broadcastOn([ DialogChannels.dialogFocusShiftedChannel ], {
```

File: modules/tinymce/src/themes/silver/main/ts/ui/dialog/IFrame.ts [hunk 5]
```diff
@@ -1,19 +1,24 @@
...
 import { Cell, Fun, Optional, Throttler, Type } from '@ephox/katamari';
 import { PlatformDetection } from '@ephox/sand';
-import { Attribute, SugarElement } from '@ephox/sugar';
+import { Attribute, Class, Compare, SugarElement, Traverse } from '@ephox/sugar';
 
 import { UiFactoryBackstageProviders } from '../../backstage/Backstage';
 import { renderFormFieldWith, renderLabel } from '../alien/FieldLabeller';
 import { RepresentingConfigs } from '../alien/RepresentingConfigs';
 import * as NavigableObject from '../general/NavigableObject';
+import { dialogFocusShiftedChannel } from '../window/DialogChannels';
```
Comment (mkzhx) [inline modules/tinymce/src/themes/silver/main/ts/ui/dialog/IFrame.ts:11]: Same here with scoped import.

File: modules/tinymce/src/themes/silver/main/ts/ui/window/SilverInlineDialog.ts [hunk 6]
```diff
@@ -11,7 +11,7 @@ import { UiFactoryBackstage } from '../../backstage/Backstage';
 import { RepresentingConfigs } from '../alien/RepresentingConfigs';
 import { formCloseEvent } from '../general/FormEvents';
```
Comment (mkzhx) [inline modules/tinymce/src/themes/silver/main/ts/ui/window/SilverInlineDialog.ts:12]: Might as well change all these to scoped imports

**Review by mkzhx — APPROVED** [review]
Great work figuring this out!

**Review by TheSpyder — APPROVED** [review]

**Pull request merged**